[BUILD] issue on aarch64 neon asm enabled on arm32

**Describe the bug**
When building a dependency with armv7 neon enabled (-fpu=neon), oiio fail with the following error:

```
In file included from /builddir/build/BUILD/olive-41a49c4880e21b11b08de94d2df1b227bf219fc6/app/codec/ffmpeg/ffmpegdecoder.cpp:32:
In file included from /usr/include/OpenImageIO/imagebuf.h:16:
In file included from /usr/include/OpenImageIO/fmath.h:45:
/usr/include/OpenImageIO/simd.h:4753:12: error: use of undeclared identifier 'vaddvq_s32'; did you mean 'vpaddlq_s32'?
    return vaddvq_s32(v);
           ^
/usr/lib/clang/13.0.1/include/arm_neon.h:18534:16: note: 'vpaddlq_s32' declared here
__ai int64x2_t vpaddlq_s32(int32x4_t __p0) {
```

**To Reproduce**
Steps to reproduce the behavior:
1. build (or cross-compile) oiio on armv7 with neon enabled.


**Expected behavior**
Build should succeed, but fails because vaddvq_s32 is only available on aarch64 and not on armv7.
Most of the time armv7 (distro/)users don't explicitly enable neon.

**Evidence**
vaddvq_s32 is tagged as aarch64 only in this documentation.
https://arm-software.github.io/acle/neon_intrinsics/advsimd.html

**Platform information:**
 - OIIO branch/version: current tag 
 - OS: Linux
 - C++ compiler: clang 13 on Fedora 36
 - Any non-default build flags when you build OIIO: -fpu=neon

--- a/src/include/OpenImageIO/simd.h
+++ b/src/include/OpenImageIO/simd.h
@@ -4751,7 +4751,7 @@ OIIO_FORCEINLINE vint4 vreduce_add (const vint4& v) {
 OIIO_FORCEINLINE int reduce_add (const vint4& v) {
 #if OIIO_SIMD_SSE
     return extract<0> (vreduce_add(v));
-#elif OIIO_SIMD_NEON
+#elif OIIO_SIMD_NEON && defined(__aarch64__)
     return vaddvq_s32(v);
 #else
     SIMD_RETURN_REDUCE (int, 0, r += v[i]);
@@ -7015,7 +7015,7 @@ OIIO_FORCEINLINE const vfloat4 & vfloat4::operator*= (float val) {
 OIIO_FORCEINLINE vfloat4 operator/ (const vfloat4& a, const vfloat4& b) {
 #if OIIO_SIMD_SSE
     return _mm_div_ps (a.m_simd, b.m_simd);
-#elif OIIO_SIMD_NEON
+#elif OIIO_SIMD_NEON && defined(__aarch64__)
     return vdivq_f32 (a.m_simd, b.m_simd);
 #else
     SIMD_RETURN (vfloat4, a[i] / b[i]);
@@ -7025,7 +7025,7 @@ OIIO_FORCEINLINE vfloat4 operator/ (const vfloat4& a, const vfloat4& b) {
 OIIO_FORCEINLINE const vfloat4 & vfloat4::operator/= (const vfloat4& a) {
 #if OIIO_SIMD_SSE
     m_simd = _mm_div_ps (m_simd, a.m_simd);
-#elif OIIO_SIMD_NEON
+#elif OIIO_SIMD_NEON && defined(__aarch64__)
     m_simd = vdivq_f32 (m_simd, a.m_simd);
 #else
     SIMD_DO (m_val[i] /= a[i]);
@@ -7036,7 +7036,7 @@ OIIO_FORCEINLINE const vfloat4 & vfloat4::operator/= (const vfloat4& a) {
 OIIO_FORCEINLINE const vfloat4 & vfloat4::operator/= (float val) {
 #if OIIO_SIMD_SSE
     m_simd = _mm_div_ps (m_simd, _mm_set1_ps(val));
-#elif OIIO_SIMD_NEON
+#elif OIIO_SIMD_NEON && defined(__aarch64__)
     m_simd = vdivq_f32 (m_simd, vfloat4(val));
 #else
     SIMD_DO (m_val[i] /= val);
@@ -7278,7 +7278,7 @@ OIIO_FORCEINLINE vfloat4 vreduce_add (const vfloat4& v) {
 OIIO_FORCEINLINE float reduce_add (const vfloat4& v) {
 #if OIIO_SIMD_SSE
     return _mm_cvtss_f32(vreduce_add (v));
-#elif OIIO_SIMD_NEON
+#elif OIIO_SIMD_NEON && defined(__aarch64__)
     return vaddvq_f32(v);
 #else
     return v[0] + v[1] + v[2] + v[3];

